Drop testing of dicts table

According to https://github.com/scylladb/scylla-enterprise/pull/4570, the dicts table is being dropped.
It has been spotted by our scylla-enterprise-nightly:latest-enterprise tests:
```
2024-08-28T10:54:39.3250678Z           	... // 1 ignored and 2 identical elements
2024-08-28T10:54:39.3251175Z           	{
2024-08-28T10:54:39.3251594Z           		... // 1 ignored and 1 identical fields
2024-08-28T10:54:39.3252851Z           		Tables: []restore.Table(Inverse(cmpopts.SortSlices, []restore.Table{
2024-08-28T10:54:39.3254161Z         - 			{Table: "dicts", TombstoneGC: "timeout"},
2024-08-28T10:54:39.3254925Z           			... // 1 ignored and 1 identical elements
2024-08-28T10:54:39.3255646Z           		})),
2024-08-28T10:54:39.3256012Z           	},
2024-08-28T10:54:39.3257254Z           	{Keyspace: "system_replicated_keys", Tables: {{Table: "encrypted_keys", TombstoneGC: "timeout"}}},
2024-08-28T10:54:39.3260688Z           	{Keyspace: "system_traces", Tables: Inverse(cmpopts.SortSlices, []restore.Table{{Table: "events", TombstoneGC: "timeout"}, {Table: "node_slow_log", TombstoneGC: "timeout"}, {Table: "node_slow_log_time_idx", TombstoneGC: "timeout"}, {Table: "sessions", TombstoneGC: "timeout"}, ...})},
2024-08-28T10:54:39.3262461Z           }))
2024-08-28T10:54:39.3262719Z         
2024-08-28T10:54:39.3263330Z --- FAIL: TestRestoreGetTargetUnitsViewsIntegration (27.74s)
2024-08-28T10:54:39.3264541Z     --- PASS: TestRestoreGetTargetUnitsViewsIntegration/tables (0.22s)
2024-08-28T10:54:39.3265944Z     --- PASS: TestRestoreGetTargetUnitsViewsIntegration/schema (0.22s)
2024-08-28T10:54:39.3266761Z     --- FAIL: TestRestoreGetTargetUnitsViewsIntegration/default_values (0.17s)
2024-08-28T10:54:39.3268387Z     --- FAIL: TestRestoreGetTargetUnitsViewsIntegration/continue_false (0.18s)
```

--- a/pkg/service/restore/service_restore_integration_test.go
+++ b/pkg/service/restore/service_restore_integration_test.go
@@ -322,9 +322,6 @@ func TestRestoreGetTargetUnitsViewsIntegration(t *testing.T) {
 			if checkAnyConstraint(t, h.Client, "< 1000") {
 				ignoreUnits = append(ignoreUnits, "system_replicated_keys")
 			}
-			if checkAnyConstraint(t, h.Client, "< 2024.2") {
-				ignoreUnits = append(ignoreUnits, "dicts")
-			}
 			if checkAnyConstraint(t, h.Client, ">= 6.0, < 2000", ">= 2024.2, > 1000") {
 				ignoreUnits = append(ignoreUnits,
 					"system_auth",

--- a/pkg/service/restore/testdata/get_target/continue_false.units.json
+++ b/pkg/service/restore/testdata/get_target/continue_false.units.json
@@ -72,11 +72,6 @@
         "size": 185891
       },
       {
-        "table": "dicts",
-        "tombstone_gc": "timeout",
-        "size": 0
-      },
-      {
         "table": "service_levels",
         "tombstone_gc": "timeout",
         "size": 0

--- a/pkg/service/restore/testdata/get_target/default_values.units.json
+++ b/pkg/service/restore/testdata/get_target/default_values.units.json
@@ -72,11 +72,6 @@
         "size": 185891
       },
       {
-        "table": "dicts",
-        "tombstone_gc": "timeout",
-        "size": 0
-      },
-      {
         "table": "service_levels",
         "tombstone_gc": "timeout",
         "size": 0

